apiserver/handler.go

```diff
@@ -0,0 +1,226 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"istio.io/manager/model"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/golang/glog"
+)
+
+// GetConfig retrieves the config object from the configuration registry
+func (api *API) GetConfig(request *restful.Request, response *restful.Response) {
+
+	params := request.PathParameters()
+	key, err := setup(params)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	proto, ok := api.registry.Get(key)
+	if !ok {
+		errLocal := &model.ItemNotFoundError{Key: key}
+		api.writeError(http.StatusNotFound, errLocal.Error(), response)
+		return
+	}
+
+	var schema model.ProtoSchema
+	retrieved, err := schema.ToJSON(proto)
+	if err != nil {
+		api.writeError(http.StatusInternalServerError, err.Error(), response)
+		return
+	}
+	var retJSON interface{}
+	if err = json.Unmarshal([]byte(retrieved), &retJSON); err != nil {
+		api.writeError(http.StatusInternalServerError, err.Error(), response)
+		return
+	}
+	config := Config{
+		Name: params["name"],
+		Type: params["kind"],
+		Spec: retJSON,
+	}
+	if err = response.WriteHeaderAndEntity(http.StatusOK, config); err != nil {
+		api.writeError(http.StatusInternalServerError, err.Error(), response)
+	}
+}
+
+// AddConfig creates and stores the passed config object in the configuration registry
+// It is equivalent to a restful PUT and is idempotent
+func (api *API) AddConfig(request *restful.Request, response *restful.Response) {
+
+	// ToDo: check url name matches body name
+
+	params := request.PathParameters()
+	key, err := setup(params)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	config := &Config{}
+	err = request.ReadEntity(config)
```

Comment:
Combine `err` assignment and check on single line here and lines 72, 78, 107, 113, 119, and 145.